lib/bolt/cli.rb

```diff
@@ -432,6 +449,18 @@ def handle_parser_errors
       raise Bolt::CLIError, "Unknown argument '#{e.args.first}'"
     end
 
+    def puppetdb_client
+      return @puppetdb_client if @puppetdb_client
+      puppetdb_config = Bolt::PuppetDB::Config.new(nil, config)
```

Comment:
This seems to result in `Could not retrieve targets from PuppetDB: no implicit conversion of Bolt::Config into Hash`

Reply:
It looks like you should be able to explicitly call `to_h` on a struct and get a hash.

Reply:
Whoops, bad rebase. That should be `config.puppetdb`...

Reply:
Yeah, that makes sense.